### PR TITLE
fix: Include parent group node when copying ForLoop nodes

### PR DESCRIPTION
**Pull request by akre54**
## Summary
Fixed copy/paste behavior for ForLoop nodes to properly include and recreate the parent group node. When pasting ForLoop nodes (e.g., ForLoopBeginOp, ForLoopEndOp), the parent group node is now automatically included in the copy set, and pasted nodes have their parentId correctly remapped to reference the new group.

## Changes
- **doCopy**: Auto-includes parent group nodes when copying child nodes
- **doPaste**: Implements two-pass ID generation with proper parentId remapping
- Added 5 new tests for ForLoop copy/paste scenarios covering edge cases

## Test Plan
- [x] All 31 existing node operation tests pass
- [x] All 1090 total tests pass
- [x] Linting passes with no warnings

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by chrisgervang — COMMENTED** [review]
This one has a code smell

**Comment by akre54** [comment]
Sorry what do you mean

This was fixing a bug

File: noodles-editor/src/noodles/__tests__/node-operations-integration.test.tsx [hunk 1]
```diff
@@ -275,11 +275,39 @@ function renameNode(
...
+
+  for (let i = 1; i < 100_000; i++) {
+    const candidatePath = pathPrefix === '/' ? `/${baseName}-${i}` : `${pathPrefix}${baseName}-${i}`
+    if (!existingNodeIds.has(candidatePath) && !hasOp(candidatePath)) {
+      return candidatePath
+    }
+  }
+
+  return newId // Fallback
+}
```
[the remark this answers is not in the document]
Comment (akre54) [reply noodles-editor/src/noodles/__tests__/node-operations-integration.test.tsx]: @copilot follow-up and remove the test helper
